subsys/bluetooth/shell/bt.c

```diff
@@ -1147,14 +1152,85 @@ static int cmd_scan_filter_clear_addr(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_OBSERVER */
 
 #if defined(CONFIG_BT_BROADCASTER)
-static const struct bt_data ad_discov[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-};
+static void ad_mode_parse(const struct shell *sh, size_t argc, char *argv[], bool *discov)
+{
+	for (size_t argn = 0; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "discov")) {
+			*discov = true;
+			break;
+		} else if (!strcmp(arg, "non_discov")) {
+			*discov = false;
+			break;
+		}
+	}
+}
+
+static ssize_t ad_init(struct bt_data *data, size_t data_size, bool discoverable)
+{
+	static uint8_t ad_flags;
+	size_t len = 0;
+	int err;
+
+	/* Set BR/EDR Not Supported if LE-only device */
+	ad_flags = IS_ENABLED(CONFIG_BT_BREDR) ? 0 : BT_LE_AD_NO_BREDR;
+
+	if (discoverable) {
+		ad_flags |= BT_LE_AD_GENERAL;
+	}
+
+	if (ad_flags != 0) {
+		__ASSERT(data_size > len, "No space for AD_FLAGS");
+		data[len].type = BT_DATA_FLAGS;
+		data[len].data_len = sizeof(ad_flags);
+		data[len].data = &ad_flags;
+		len++;
+	}
+
+#if defined(CONFIG_BT_AUDIO)
+	/* Advertise RSi in discoverable mode only */
+	if (IS_ENABLED(CONFIG_BT_CSIS) && csis != NULL && discoverable) {
```

Comment:
I think this is a decent way of adding service data to the BT shell advertise command, as it can be fairly easily extended. 

The part that is guarded by `IS_ENABLED(CONFIG_BT_CSIS)` could perhaps be moved to a function in the CSIP/CSIS shell module instead and simply called from here